### ISSUE TITLE
Hidden Columns in Column Metadata

### ISSUE THREAD

**Issue by dannyparsons** [IDEMSInternational/R-Instat#4095]
I've hidden some columns. In the grid I only see the visible columns, but in the Column Metadata I still see all columns. Should this only show the visible columns too like the Data View? I think so, it looks a bit confusing otherwise.

**Comment by rdstern** [comment]
I agree that the default should be that the columns (rows in the metadata) are hidden too.

I add related thoughts below.  My suggestion for now is as follows:

**My initial solution is to hide rows in the column metadata when columns are hidden in the data.  Then for the row numbers to go red (as in the data) when a filter is on.  And the numbers have gaps in the column metadata - which they will - when columns are hidden.  So we will know.  And the text at the bottom of the data could be extended to clarify whether columns are hidden, purposely or because we are only showing the first 30 columns.**

Of course the ideal is that you could make hidden columns visible (at least in the metadata) and then they are coloured or otherwise clearly different.
Genstat on the filter (so different) has the following feature that I quite like:

![genstat filter](https://user-images.githubusercontent.com/11226469/31998131-90da1aec-b986-11e7-918e-c7a68be39ab9.png)

It uses the top left-hand corner to show that rows are filtered.  I am happy with our method with a filter, but would also like something to be visible in the column metadata.
Then the cross in the top right-hand corner is a toggle to show or hide the filtered rows.  When shown they are in red, to indicate they are filtered.

So, if we only do one thing now on hiding, then I agree, we hide the corresponding rows in the metadata.  Ideally we have an easy way of switching what is visible without necessarily turning the past hiding on and off (or in addition to that.  

Is there another toolbar item coming - possibly with a drop down?  And will there be anything visible in the sheet to indicate that something is hidden.  I often get confused in Excel, but the lettering helps.  Here something is needed isn't it?  I notice that (at the bottom) it now says for me:

![rows and columns](https://user-images.githubusercontent.com/11226469/31998615-5d0e88f4-b988-11e7-910a-81efa142caf2.png)

So here we could indicate if some are hidden.  It sort of indicates by 5 of 7, but I am not sure what it would say if there were more than 30?  Perhaps this is enough!

So, on the metadata, it will be indicated a bit if we hide columns, because the numbers will have gaps.  That's OK for a start.

**Comment by lilyclements** [comment]
Hidden columns are no longer visible in the column metadata
